libraries/c_sdk/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -499,13 +499,17 @@ static IotMqttError_t _deserializeIncomingPacket( _mqttConnection_t * pMqttConne
 
                 if( contextIndex >= 0 )
                 {
-                    IotMutex_Lock( &( connToContext[ contextIndex ].contextMutex ) );
+                    if( IotMutex_TakeRecursive( &( connToContext[ contextIndex ].contextMutex ) ) == false )
+                    {
+                        IOT_SET_AND_GOTO_CLEANUP( IOT_MQTT_TIMEOUT );
+                    }
+
                     /* Updating the status for the outgoing publishes after the corresponding puback is received. */
                     publishStateStatus = MQTT_UpdateStateAck( &( connToContext[ contextIndex ].context ),
                                                               pIncomingPacket->packetIdentifier,
                                                               MQTTPuback,
                                                               MQTT_RECEIVE, &publishRecordState );
-                    IotMutex_Unlock( &( connToContext[ contextIndex ].contextMutex ) );
+                    IotMutex_GiveRecursive( &( connToContext[ contextIndex ].contextMutex ) );
```

Comment:
doesn't the return type need to be checked here?

Reply:
Addressed